### ISSUE TITLE
rustdoc: broken_intra_doc_links lint triggered by item behind feature gate

### ISSUE THREAD

**Issue by ctz** [rust-lang/rust#114626]
```rust
#[cfg(feature = "notset")]
pub struct BehindFeature;

/// Frobs a blorb.
/// 
/// See also [`BehindFeature`].
pub struct Hello;
```

Causes this warning:

```
warning: unresolved link to `BehindFeature`
 --> src/main.rs:6:16
  |
6 | /// See also [`BehindFeature`].
  |                ^^^^^^^^^^^^^ no item named `BehindFeature` in scope
  |
  = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
  = note: `#[warn(rustdoc::broken_intra_doc_links)]` on by default

warning: `testcrate` (bin "testcrate" doc) generated 1 warning
```

(and, of course, `cargo doc --all-features` doesn't, and reveals that the link is not broken.)

Is it feasible to make this lint notice that the link is to an item behind a feature gate? And, either include that in the warning, or even elide the lint altogether? Happy to do the legwork if the rustdoc maintainers think it is feasible.

I think the compiler is doing something similar when it reports "found an item that was configured out" in errors, eg:

```
error[E0432]: unresolved import `super::BehindFeature`
  --> src/differentcodetoabove.rs:18:13
   |
18 |         use super::BehindFeature;
   |             ^^^^^^^^^^^^^^^^^^^^ no `BehindFeature` in the root
   |
note: found an item that was configured out
  --> src/differentcodetoabove.rs:2:12
   |
2  | pub struct BehindFeature;
   |            ^^^^^^^^^^^^^
   = note: the item is gated behind the `notset` feature
```

**Comment by GuillaumeGomez** [comment]
rustdoc relies on rustc's information to generate intra doc links. If the item is cfg-ed out, then it's not possible. Changing this would require huge changes so it's very likely not going to happen anytime soon.

**Comment by JensMertelmeyer** [comment]
Thank you for letting us know about the `--all-features` flag. Better than nothing.
